documentation/docs/pkce_authentication.md

```diff
@@ -0,0 +1,145 @@
+# Using PKCE Authentication with KMS
+
+This document provides a comprehensive guide on using PKCE (Proof Key for Code Exchange) authentication with the Cosmian KMS.
+
+## Overview
+
+PKCE enhances the security of OAuth 2.0 authentication flows by eliminating the need for client secrets. This makes it particularly valuable for:
+
+- Public clients that cannot securely store client secrets
+- Mobile applications
+- Single-page applications
+- Desktop applications
+- CLI tools
+
+## Benefits of PKCE
+
+- **Improved Security**: Eliminates the need to store and manage client secrets
+- **Simplified Deployment**: Reduces configuration complexity for public clients
+- **Protection Against CSRF & Authorization Code Interception**: The code verifier provides additional security
+- **Standards Compliance**: Follows OAuth 2.0 and OpenID Connect standards
+
+## How PKCE Works
+
+1. **Authorization Request**:
+   - Client generates a random code verifier and code challenge
+   - Client requests authorization with the code challenge
+   - User authenticates and server issues an authorization code
+
+2. **Token Exchange**:
+   - Client exchanges the authorization code and code verifier for tokens
+   - Server verifies that the code verifier matches the challenge
+   - Server issues access and ID tokens upon successful verification
+
+## Implementation Details
+
+### Client-Side Changes
+
+The KMS client library now supports PKCE authentication with optional client secrets:
+
+
+**Example: Entra ID**
```

Comment:
Since Google is currently the only Identity Provider that still requires a `client_secret`, it would be helpful to include a specific example using Google and explicitly mention this requirement

Reply:
I didn't intend to make them look bad, but wth, I added a note.